--- mDNSResponder-1310.140.1/mDNSShared/dnsextd_parser.y.orig
+++ mDNSResponder-1310.140.1/mDNSShared/dnsextd_parser.y
@@ -23,7 +23,7 @@
 #include "DebugServices.h"
 #include "dnsextd.h"
 
-void yyerror( const char* error );
+void yyerror( void *context, const char* error );
 int  yylex(void);
 
 
@@ -131,6 +131,8 @@
 %type	<string>	addressstatement
 %type	<string>	networkaddress
 
+%parse-param {void *context}
+
 %%
 
 commands:
@@ -409,7 +411,7 @@
 
 extern int yylineno;
 
-void yyerror( const char *str )
+void yyerror( void *context, const char *str )
 {
         fprintf( stderr,"%s:%d: error: %s\n", g_filename, yylineno, str );
 }
